service/discovery/azure/key_vault.go

```diff
@@ -353,6 +353,7 @@ func (d *azureKeyVaultDiscovery) getKeys(kv *armkeyvault.Vault) ([]*voc.Key, err
 	if c == nil {
 		return nil, errors.New("keys client is empty")
 	}
+	// TODO(all): We should use the listPager function, otherwise the call does not work if the Clouditor starts without a specific resource group.
```

Comment:
Doesn't work here because the list functions are different from the generic ones (here you only have one because a resource group always have to be provided as well). Anyway that is a different issuer and PR...

Reply:
And yes it works if you replace `(util.Deref(d.azureDiscovery.rg` with sth. like `resourceGroup(util.Deref(kv.name))`. but again, different issue+PR

Reply:
Regarding the first comment: You're right, we would have to use the allPages function. Maybe, we could think about exporting the allPages function and use that in the future.